sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -361,7 +361,19 @@ def test_piecewise_simplify():
     p = Piecewise(((x**2 + 1)/x**2, Eq(x*(1 + x) - x**2, 0)),
                   ((-1)**x*(-1), True))
     assert p.simplify() == \
-        Piecewise((1 + 1/x**2, Eq(x, 0)), ((-1)**(x + 1), True))
+        Piecewise((zoo, Eq(x, 0)), ((-1)**(x + 1), True))
```

Comment:
Wrong result?

Reply:
simplification now substitutes known values of variables contained in the condition. `(1/x).subs(x,0)` gives `zoo` so `zoo` is now returned. Perhaps it would be better to leave the expression instead of the substituted form if any `NaN` or `zoo` is introduced?

Reply:
That seems like a reasonable change. I guess we don't hold the outputs of simplify to any kind of strict solution, since it is impossible to define the simplest solution. No change needed. Just checking for potential backwards compatibility issues. Changed tests always raise flags for me.